CMakeLists.txt

```diff
@@ -286,7 +287,24 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 # CUDA compiler
 if(WITH_CUDA)
-  find_package(CUDACompiler REQUIRED)
+  set(CMAKE_CUDA_STANDARD 9.0)
+  set(WITH_CUDA_COMPILER "nvcc" CACHE STRING "Select the CUDA compiler, can be 'nvcc' (default), 'hip', or 'clang'")
+  if (WITH_CUDA_COMPILER STREQUAL "nvcc")
+    find_package(CUDA ${CMAKE_CUDA_STANDARD} REQUIRED)
+    find_package(CUDACompilerNVCC ${CMAKE_CUDA_STANDARD} REQUIRED)
+  elseif (WITH_CUDA_COMPILER STREQUAL "hip")
+    set(ROCM_HOME "/opt/rocm" CACHE FILEPATH "Path to AMD ROCm")
+    list(APPEND CMAKE_MODULE_PATH "${ROCM_HOME}/hip/cmake")
+    find_package(HIP 1.5.18494 MODULE REQUIRED)
+    set(HCC_PATH "${HIP_ROOT_DIR}")
+    find_package(HIP 1.5.18494 MODULE REQUIRED)
```

Comment:
why do you have to call ```find_package(HIP``` two times?

Reply:
it's explained in 9ac0c09e7

Reply:
I've added an extra CMake comment on top of the git commit explanation.